src/couch/test/eunit/couchdb_file_compression_tests.erl

```diff
@@ -40,58 +67,34 @@ setup_all() ->
     ),
     {ok, _} = couch_db:update_doc(Db, DDoc, []),
     ok = couch_db:close(Db),
-    {Ctx, DbName}.
+    ok = refresh_index(DbName),
+    DbName.
 
-teardown_all({Ctx, DbName}) ->
-    ok = couch_server:delete(DbName, [?ADMIN_CTX]),
-    test_util:stop_couch(Ctx).
+teardown(DbName) ->
+    config:delete("couchdb", "file_compression", false),
+    couch_server:delete(DbName, [?ADMIN_CTX]).
 
-couch_file_compression_test_() ->
-    {
-        "CouchDB file compression tests",
-        {
-            setup,
-            fun setup_all/0,
-            fun teardown_all/1,
-            {with, [
-                fun should_use_none/1,
-                fun should_use_deflate_1/1,
-                fun should_use_deflate_9/1,
-                fun should_use_snappy/1,
-                fun should_compare_compression_methods/1
-            ]}
-        }
-    }.
+should_use_none(DbName) ->
+    config:set("couchdb", "file_compression", "none", false),
+    compact_db(DbName),
+    compact_view(DbName).
 
-should_use_none({_, DbName}) -> run_test(DbName, "none").
-should_use_deflate_1({_, DbName}) -> run_test(DbName, "deflate_1").
-should_use_deflate_9({_, DbName}) -> run_test(DbName, "deflate_9").
-should_use_snappy({_, DbName}) -> run_test(DbName, "snappy").
-
-should_compare_compression_methods({_, DbName}) ->
-    TestDb = setup_db(DbName),
-    Name = "none > snappy > deflate_1 > deflate_9",
-    try
-        {Name, {timeout, ?TIMEOUT, ?_test(compare_methods(TestDb))}}
-    after
-        couch_server:delete(TestDb, [?ADMIN_CTX])
-    end.
+should_use_deflate_1(DbName) ->
+    config:set("couchdb", "file_compression", "deflate_1", false),
+    compact_db(DbName),
+    compact_view(DbName).
 
-run_test(DbName, Comp) ->
-    config:set("couchdb", "file_compression", Comp, false),
-    Timeout = 5 + ?TIMEOUT,
-    TestDb = setup_db(DbName),
-    Tests = [
-        {"compact database", {timeout, Timeout, ?_test(compact_db(DbName))}},
-        {"compact view", {timeout, Timeout, ?_test(compact_view(DbName))}}
-    ],
-    try
-        {"Use compression: " ++ Comp, Tests}
-    after
-        ok = couch_server:delete(TestDb, [?ADMIN_CTX])
-    end.
```

Comment:
Very fun to wrap one's head around this :)

Reply:
This might have been needed before we had the setup / teardown pattern with setup/foreach.